src/domain/accounts/entities/upsert-account-data-settings.dto.entity.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from 'zod';
+
+export class UpsertAccountDataSettingsDto
+  implements z.infer<typeof UpsertAccountDataSettingsDtoSchema>
+{
+  accountDataSettings: {
+    dataTypeName: string;
```

Comment:
Will the client retrieve these "data types" from us? If so, it might be better to upsert them via ID. What do you think?

A small nit: I don't think we need prefix this with "dataType-" either.

Reply:
Initially, I thought about using ids instead of names for this payload. I opted for the names to make it more meaningful for the clients, but I agree it's simpler to implement this endpoint if we change back to ids. Wdyt?

Reply:
I think IDs would make sense as they remain static whereas names could be subject to change. I'm happy to merge this PR and change it in a follow-up though. (Not sure if it'd cause issues due to migrations though.)

Reply:
IDs make total sense to me also, they were my first option as well, but later I took this approach to make it easier for the clients. But giving it a second thought, I agree it makes sense to have IDs instead of names to stay static. I will change it in this PR! Thanks! 😄